### DISCUSSION TITLE
Why useFormState vs formState?

### DISCUSSION THREAD

**Discussion by GollyJer** [react-hook-form/react-hook-form#4699] [Q&A] [answered]
Just upgraded to v7. Most of the issues/bugs I had with v6 related to `defaultValues`. I'm happy to report they are all fixed! 🚀

My question now is why would one choose to use `userFormState` vs `formState` from `useForm`?
Is it purely if you have multiple forms in a large application? In that case why wouldn't accessing `formState` from each individual form provide the same level of optimization.

I'm clearly missing something here so I thought I'd ask.

My entire usage of react-hook-forms is on single-screen, single-form, react-native screens with the Controller component.
Can I benefit from `userFormState`?

Thanks!

**Comment by bluebill1049** [comment, marked as answer]
`useFormState` is at the component level, doesn't trigger re-render at the root. I think we should make this clearer in the doc.

The same applies to `watch` vs `useWatch`. it's better to use those hooks if you have a large and complex forms.

**Reply by bluebill1049** [reply]
Take a look at this example: https://codesandbox.io/s/calc-i231d?file=/src/App.tsx which i have used useWatch and the caculation re-render will only happen at the <Total /> component, leave the root component peacefully.

**Comment by zhixuan-loh** [comment]
If we are accessing `formState` from  a `useFormContext`, will `useFormState` give me performance improvement?

**Reply by hjiog** [reply]
I think it's not possible. Once control._subjects.state change, it will trigger useForm change, so I think useFormState and formState produce the same effect.
